backend/src/controllers/loginController.ts

```diff
@@ -0,0 +1,63 @@
+import { Request, Response } from "express";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
+import { Farmer } from "../models/farmer";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+interface LoginRequestBody {
+  email?: string;
+  password?: string;
+}
+
+const loginController = async (
+  req: Request<{}, {}, LoginRequestBody>,
+  res: Response
+) => {
+  const { email, password } = req.body;
+
+  // Validate credentials
+  if (!email || !password) {
+    return res.status(400).json({ message: "Email and password are required" });
+  }
+
+  try {
+    // Check if farmer exists
+    const farmer = await Farmer.findOne({ email }).select("+password");
+    if (!farmer) {
+      return res.status(401).json({ message: "Invalid email or password" });
+    }
+
+    // Check if password matches
+    const isMatch = await bcrypt.compare(password, farmer.password);
+    if (!isMatch) {
+      return res.status(401).json({ message: "Invalid email or password" });
+    }
+
+    // Ensure JWT_SECRET is available
```

Comment:
This could be the first check or even a separate Middleware

Reply:
Do you mean we should divide the email+password verification and issuing cookie and JST into two controllers?